tests/publiccloud/check_registercloudguest.pm

```diff
@@ -26,18 +26,17 @@ my $path = is_sle('=15-SP2') ? '/usr/sbin/' : '';    # 15-SP2 is the oldest vers
 my $regcode_param = (is_byos()) ? "-r " . get_required_var('SCC_REGCODE') : '';
 
 sub run {
-    my ($self, $args) = @_;
-    my ($provider, $instance);
+    my ($self) = @_;
+    my ($instance);
     select_host_console();
 
     $run_count++;
 
     if (get_var('PUBLIC_CLOUD_QAM', 0)) {
```

Comment:
we don't need this whole if/else block anymore ! the whole point of this block was that we have cases where provider and instance are created in different place and just passed here through run_args and in other cases we created them directly here . Now we can just move initialization in `lib/publiccloud/basetest` and all tests will always act like instance is already there !